client/src/components/Submenu/index.js

```diff
@@ -0,0 +1,42 @@
+/**
+ * Initialises the Submenus within the primary Wagtail menu (excluding the Explorer menu)
+ */
+
+const initSubmenus = () => {
+  const primaryNavContainer = document.querySelector('[data-nav-primary]');
+
+  if (!primaryNavContainer) {
+    return;
+  }
+
+  const subMenus = document.querySelectorAll('[data-nav-primary-submenu-trigger]');
+  const activeClass = 'submenu-active';
+
+  subMenus.forEach(subMenu => {
+    subMenu.addEventListener('click', clickEvent => {
+      primaryNavContainer.classList.remove(activeClass);
+      subMenus.forEach(sm => sm.classList.remove(activeClass));
+
+      primaryNavContainer.classList.toggle(activeClass);
+      subMenu.parentNode.classList.toggle(activeClass);
+
+      document.addEventListener('mousedown', e => {
+        if (!subMenu.contains(e.target) && subMenu !== e.target) {
+          primaryNavContainer.classList.remove(activeClass);
+          subMenu.parentNode.classList.remove(activeClass);
+        }
+      });
+
+      document.addEventListener('keydown', e => {
+        if (e.keyCode === 27) {
```

Comment:
minor item - for the sake of readability it might be nicer to use our core variables for the esc key

wagtail/admin/static_src/wagtailadmin/js/core.js
`wagtail.keys.ESC` - or something like that.

I know the previous code did not do this but just aids in understanding

Reply:
I wasn't aware of this variable - thanks! Although I have just used `event.key` in another project, and that gives you a human readable value (eg `Enter`, `Shift`). This looks to be IE9+ - I might quickly look further at that and see if there are any caveats there. If that works out we might be able to remove the `wagtail.keys` :)

Reply:
I've used `event.key` here, annoyingly I have to check against 2 strings  - but it's still more readable :)